src/java/picard/analysis/CollectGcBiasMetrics.java

```diff
@@ -83,187 +87,105 @@
     @Option(shortName = "BS", doc = "Whether the SAM or BAM file consists of bisulfite sequenced reads.  ")
     public boolean IS_BISULFITE_SEQUENCED = false;
 
-    // Used to keep track of the total clusters as this is kinda important for bias
-    private int totalClusters = 0;
-    private int totalAlignedReads = 0;
-    // Histograms to track the number of windows at each GC, and the number of read starts
-    // at windows of each GC. Need 101 to get from 0-100.
-    private static final int WINDOWS = 101;
-    private final int[] windowsByGc = new int[WINDOWS];
-    private final int[] readsByGc = new int[WINDOWS];
-    private final long[] basesByGc = new long[WINDOWS];
-    private final long[] errorsByGc = new long[WINDOWS];
-    private static int lastContig = SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX;
-    private byte[] gc;
-    private byte[] refBases;
-    private String saveHeader;
+    @Option(shortName = "LEVEL", doc = "The level(s) at which to accumulate metrics.  ")
+    private final Set<MetricAccumulationLevel> METRIC_ACCUMULATION_LEVEL = CollectionUtil.makeSet(MetricAccumulationLevel.ALL_READS);
 
-    /** Stock main method. */
+    // Calculates GcBiasMetrics for all METRIC_ACCUMULATION_LEVELs provided
+    private GcBiasMetricsCollector multiCollector;
+    public byte[] refBases;
+    public final int windowSize = WINDOW_SIZE;
+    final int[] windowsByGc = new int[WINDOWS];
+    public static final int WINDOWS = 101;
```

Comment:
This is a little confusing, but the number of WINDOWS should always be 101 because we want gcbias calculated from 0% gc content to 100% gc content. However, WINDOW_SIZE is a command line input that corresponds to the size of the window that moves along the reference genome. WINDOW_SIZE doesn't always need to be 100 but WINDOWS should always be 101.